cli/src/main/kotlin/ls/kafka/backup/s3/BackupFile.kt

```diff
@@ -1,28 +1,10 @@
 package ls.kafka.backup.s3
 
-import ls.kafka.model.DumpRecord
-
-/**
- * Represents a single backup file in S3.
- */
-data class BackupFile(
-    val key: String,                // S3 object key
-    val records: List<DumpRecord>   // records in the file
-) {
-    val topic: String       // corresponds to the Kafka topic from which the records were read
-    val partition: Int      // Kafka partition number from which the records were read
-    val startOffset: Long   // offset of the first record written to this file
-
-    init {
-        // key format is: <prefix>/../<topic>+<kafkaPartition>+<startOffset>.<format>
-        key.split("/").last().substringBefore(".").split("+").let { nameParts ->
-            topic = nameParts[0]
-            partition = nameParts[1].toInt()
-            startOffset = nameParts[2].toLong()
-        }
-    }
-
-    override fun toString(): String {
-        return "BackupFilePath(name='$key', topic='$topic', partition=$partition, startOffset=$startOffset, recordSize=${records.size})"
+fun parseBackupFileKey(key: String): Triple<String, Int, Long> =
```

Comment:
Maybe consider creating dataclass instead of triple. I think it would make the code where the keys are filtered easier to verify if there are named properties instead of `first, second, third`.
https://github.com/lovelysystems/lovely-kafka-backup/blob/mibe/format-improvement/cli/src/main/kotlin/ls/kafka/backup/s3/S3ClientExt.kt#L40